bin/utest/cred.c

```diff
@@ -1,155 +1,130 @@
-  assert(error == 0);
+  syscall_ok(setresuid(ruid, euid, suid));
 
-  getresuid(&ruid, &euid, &suid);
+  syscall_ok(getresuid(&ruid, &euid, &suid));
   assert(ruid == 1 && euid == 2 && suid == 3);
 
   /* we can only change to value that is one of real, effective or saved */
   ruid = -1, euid = 3, suid = -1;
-  error = setresuid(ruid, euid, suid);
-  assert(error == 0);
+  syscall_ok(setresuid(ruid, euid, suid));
 
-  getresuid(&ruid, &euid, &suid);
+  syscall_ok(getresuid(&ruid, &euid, &suid));
   assert(ruid == 1 && euid == 3 && suid == 3);
 
   /* we cannnot change to value that is not one of real, effective or saved */
   ruid = -1, euid = 2, suid = -1;
-  error = setresuid(ruid, euid, suid);
-  assert(error < 0 && errno == EPERM);
+  syscall_fail(setresuid(ruid, euid, suid), EPERM);
 
-  getresuid(&ruid, &euid, &suid);
+  syscall_ok(getresuid(&ruid, &euid, &suid));
   assert(ruid == 1 && euid == 3 && suid == 3);
 
   return 0;
 }
 
-int test_get_set_gid(void) {
+TEST_ADD(get_set_gid) {
   gid_t rgid, egid, sgid;
-  int error;
 
   /* check if nothing fail if we put wrong addresses */
-  error = getresgid((void *)1, (void *)1, (void *)1);
-  assert(error < 0 && errno == EFAULT);
+  syscall_fail(getresgid((void *)1, (void *)1, (void *)1), EFAULT);
 
-  error = getresgid(&rgid, &egid, &sgid);
-  assert(error == 0);
+  syscall_ok(getresgid(&rgid, &egid, &sgid));
 
   /* assume we are running tests as root */
   assert(rgid == 0 && egid == 0 && sgid == 0);
 
   /* as root we chan change id to ony other */
   rgid = 1, egid = 2, sgid = 3;
-  error = setresgid(rgid, egid, sgid);
-  assert(error == 0);
+  syscall_ok(setresgid(rgid, egid, sgid));
 
-  getresgid(&rgid, &egid, &sgid);
+  syscall_ok(getresgid(&rgid, &egid, &sgid));
   assert(rgid == 1 && egid == 2 && sgid == 3);
 
   /* dropping privileges */
-  setresuid(1, 1, 1);
+  syscall_ok(setresuid(1, 1, 1));
 
   /* we can only change to value that is one of real, effective or saved */
   rgid = -1, egid = 3, sgid = -1;
-  error = setresgid(rgid, egid, sgid);
-  assert(error == 0);
+  syscall_ok(setresgid(rgid, egid, sgid));
 
-  getresgid(&rgid, &egid, &sgid);
+  syscall_ok(getresgid(&rgid, &egid, &sgid));
   assert(rgid == 1 && egid == 3 && sgid == 3);
 
   /* we cannnot change to value that is not one of real, effective or saved */
   rgid = -1, egid = 2, sgid = -1;
-  error = setresgid(rgid, egid, sgid);
-  assert(error < 0 && errno == EPERM);
+  syscall_fail(setresgid(rgid, egid, sgid), EPERM);
 
-  getresgid(&rgid, &egid, &sgid);
+  syscall_ok(getresgid(&rgid, &egid, &sgid));
   assert(rgid == 1 && egid == 3 && sgid == 3);
 
   return 0;
 }
 
-int test_get_set_groups(void) {
+TEST_ADD(get_set_groups) {
   gid_t rgrp[NGROUPS_MAX], gidset[NGROUPS_MAX] = {0, 1, 2, 3, 4, 5};
-  int r, ngroups = 6;
+  const int ngroups = 6;
   uid_t euid;
 
   /* check if we are a root at start */
   getresuid(NULL, &euid, NULL);
+  // TODO(fzdob): this fails with EFAULT
```

Comment:
@franciscozdo Why this function fails with EFAULT?

Reply:
I can make an PR that fixes that return value for `getresuid` (and for `getresgid` where is the same problem).

Reply:
The manual entry I had referred to: https://www.freebsd.org/cgi/man.cgi?query=getresuid&sektion=2&n=1

Reply:
Please fix it when this PR is merged. I'll leave TODO comment to be removed when the bug is fixed.